lib/functions.php

```diff
@@ -52,8 +52,8 @@ function resolve(string $basePath, string $newPath): string
     $newParts['host'] = $pick('host');
     $newParts['port'] = $pick('port');
 
-    $path = '';
-    if ($delta['path']) {
+    $path = $base['path'] ?: '/';
+    if (is_string($delta['path']) and strlen($delta['path']) > 0) {
```

Comment:
I guess we can simplify this PR to only change this line to the following:

```suggestion
    if ($delta['path'] !== null) {
```

I guess there is no need to remove the `else` block and move it before the if-statement, right?

Reply:
Done

Reply:
would the suggested change above with only checking for NULL also work for your case?

Reply:
@staabm I tried it breaks other test cases
```
...........EE.E.                                                  16 / 16 (100%)

Time: 77 ms, Memory: 4.00MB

There were 3 errors:

1) Sabre\Uri\ResolveTest::testResolve with data set #11 ('http://example.org/path.json', '#', 'http://example.org/path.json')
Uninitialized string offset: 0

/Users/somebody/Projects/uri/lib/functions.php:58
/Users/somebody/Projects/uri/tests/ResolveTest.php:16
/Users/somebody/Projects/uri/vendor/phpunit/phpunit/phpunit:53

2) Sabre\Uri\ResolveTest::testResolve with data set #12 ('http://www.example.com', '#', 'http://www.example.com/')
Uninitialized string offset: 0

/Users/somebody/Projects/uri/lib/functions.php:58
/Users/somebody/Projects/uri/tests/ResolveTest.php:16
/Users/somebody/Projects/uri/vendor/phpunit/phpunit/phpunit:53

3) Sabre\Uri\ResolveTest::testResolve with data set #14 ('http://example.org/path.json', '#', 'http://example.org/path.json')
Uninitialized string offset: 0

/Users/somebody/Projects/uri/lib/functions.php:58
/Users/somebody/Projects/uri/tests/ResolveTest.php:16
/Users/somebody/Projects/uri/vendor/phpunit/phpunit/phpunit:53

ERRORS!
Tests: 16, Assertions: 13, Errors: 3.
```

Reply:
It work if I change like this
```
if ($delta['path'] !== null && $delta['path'] !== '') {
```